### ISSUE TITLE
Golang dependencies don't seem to resolve automatically

### ISSUE THREAD

**Issue by iridian-ks** [magefile/mage#287]
## Version + Platform

I installed `mage` just now through Homebrew.

```
 mage -version
Mage Build Tool v1.9.0
Build Date: 2019-09-14T03:27:11+01:00
Commit: 1c36bf7
built with: go1.13
```

## Steps

```
mage -init
mage -l
```

## Expected Results

List the tasks from the default `magefile.go` file.

I suppose another expectation is that if the `magefile.go` is referencing dependencies that don't exist that mage would call go get to resolve that?

## Actual Results

```
mage -l
magefile.go:10:2: cannot find package "github.com/magefile/mage/mg" in any of:
        /usr/local/Cellar/go/1.13.7/libexec/src/github.com/magefile/mage/mg (from $GOROOT)
        /Users/dgilmor/go/src/github.com/magefile/mage/mg (from $GOPATH)
Error: error compiling magefiles
```

## Manual workarounds

I would have thought that this should grab all dependencies:

```
go get -d
can't load package: package .: build constraints exclude all Go files in /Users/user/proj
```

Removing the `//+build mage` tag and then re-running `go get -d` works!

Verifying:

```
mage -l
Targets:
  build          A build step that requires additional params, or platform specific steps for example
  clean          up after yourself
  install        A custom install step if you need your bin someplace other than go/bin
  installDeps    Manage your deps, or running package managers.
```

## Extra notes

I think this project is neat and very cool. I am half-way working through building a module system with GNU Make, but I wanted to try this project first. We have about a hundred different repos featuring Terraform/Ansible/Helm charts/Dockerfiles etc. I was going to abstract those build rules into modules and this is really the only build system that has a built-in module system (via go get) as well as remaining simple (conversely to Gradle which has modules, but is ridiculously hard to extend).

The idea is in our Jenkins environment we'd have the exact same build rules for every repo. Something like `mage build` followed by `mage test` and then `mage deploy`. The parity between repos would then be dealt with via mage for all of our repos, which simplifies CI.

I think where this _seems_ to fall short is in the bootstrapping story? I could just be doing something wrong, but wanted to highlight that the initial bootstrap was more difficult than anticipated.

To sum up into a succinct question:

1. Why shouldn't `mage` manage the dependencies in the `magefile`?

I'm willing to help contribute if this seems easy enough and it's a feature you'd be willing to accept.

**Comment by natefinch** [comment]
It will work correctly if you add a go.mod to the directory (one way to do it is to run `go mod init`). Otherwise it thinks you're in GOPATH mode, and looks in your gopath for the importants, and won't auto-download them if they're not there.

**Comment by iridian-ks** [comment]
Hi nate! Thanks for the swift response, I really appreciate it.

I added a `go.mod` file with these contents:

```
cat go.mod
module github.com/magefile/mage/mg

go 1.13
```

```
$ ls
go.mod  magefile.go
```

```
$ cat go.mod
module github.com/magefile/mage/mg

go 1.13
```

```
mage -l
build github.com/magefile/mage/mg: cannot load github.com/magefile/mage/mg: no Go source files
Error: error compiling magefiles
```

Moving the source files inside `$GOPATH/src` has the some results it seems.

**Comment by ntrrg** [comment]
Actually, the `go.mod` file shouldn't have `module github.com/magefile/mage/mg` because Go will try to find the Mage source files in your project directory, it should be something related to your project, or even just any name like `module myproject` if you don't will use it as a library.

**Comment by iridian-ks** [comment]
Thanks. I realized that after generating the go.mod file while being in GOPATH.

```
module mage

go 1.13

require github.com/magefile/mage v1.9.0
```

Looks like while in GOPATH things seem to work as expected.

Thanks for all the help!
